packages/tailwindcss/src/index.ts

```diff
@@ -152,28 +152,35 @@ export function compile(
       }
     }
 
-    // Drop instances of `@media reference`
+    // Drop instances of `@media theme(…)`
     //
-    // We support `@import "tailwindcss/theme" reference` as a way to import an external theme file
-    // as a reference, which becomes `@media reference { … }` when the `@import` is processed.
-    if (node.selector === '@media reference') {
+    // We support `@import "tailwindcss/theme" theme(reference)` as a way to
+    // import an external theme file as a reference, which becomes `@media
+    // theme(reference) { … }` when the `@import` is processed.
+    if (node.selector.startsWith('@media theme(')) {
+      let themeParams = node.selector.slice(13, -1)
+
       walk(node.nodes, (child) => {
         if (child.kind !== 'rule') {
           throw new Error(
-            'Files imported with `@import "…" reference` must only contain `@theme` blocks.',
+            'Files imported with `@import "…" theme(…)` must only contain `@theme` blocks.',
           )
         }
         if (child.selector === '@theme') {
-          child.selector = '@theme reference'
+          child.selector = '@theme ' + themeParams
           return WalkAction.Skip
         }
       })
       replaceWith(node.nodes)
+      return WalkAction.Skip
     }
 
-    if (node.selector !== '@theme' && node.selector !== '@theme reference') return
+    if (node.selector !== '@theme' && !node.selector.startsWith('@theme ')) return
+
+    let [_, ...themeOptions] = segment(node.selector, ' ')
 
-    let isReference = node.selector === '@theme reference'
+    let isReference = themeOptions.includes('reference')
+    let isInline = themeOptions.includes('inline')
```

Comment:
If you apply the suggestion from above and if you are worried about the `includes` that it needs to check `@media` (first argument of the array), then you can skip it by doing something like this:

```suggestion
    let isReference = themeOptions.includes('reference', 1)
    let isInline = themeOptions.includes('inline', 1)
```
Ultimate performance!

Reply:
TIL I had no idea includes took a starting position